src/main/java/frc/robot/subsystems/StateMachine.java

```diff
@@ -282,11 +273,10 @@ public Command scoreCoral() {
         () -> {
           if (manualOverride) {
             m_dragon.score().schedule();
-          } else if (m_dragon.isCoralOnDragon() && DragonState.SCORE_READY == m_dragon.getState()
+          } else if (m_dragon.isCoralOnDragon() && m_state == State.DRAGON_READY
               && ElevatorState.SCORE_READY == m_elevator.getState()) {
```

Comment:
we should allow dragon score even if coral is not detected on the dragon. This would be for cases like you try to score, but end up cancelling the action but the coral is still on the dragon but not enough to break the beam.

the `DRAGON_READY` state is a good enough guard because we can only get in that state if the coral gets onto the dragon. Additionally, we don't have to check elevator state for the same reason.

```suggestion
          } else if (m_state == State.DRAGON_READY) {
```